### ISSUE TITLE
Can crypto-js decrypt RSA/ECB/PKCS1Padding with public key?

### ISSUE THREAD

**Issue by ssaguiar** [brix/crypto-js#351]
I have a RSA (RSA/ECB/PKCS1Padding) buffer content which must be decrypted using a public key, generating a key.
This will produce the key I will need to decrypt another buffer using the AES/CBC/PKCSSPadding and an IV.

So, I will need to decrypt the first buffer using RSA and a public key to generate the key to decrypt another buffer using AES and anIV wch is in another buffer.

This is how data is present in main buffer:

main buffer bytes:

0 255 ENCRYPTED_LICENSE_KEY
256 271 IV
272 END ENCRYPTED_LICENSE_BODY

LICENSE_KEY = ENCRYPTED_LICENSE_KEY decrypted with public key (RSA/ECB/PKCS1Padding)

LICENSE_BODY = ENCRYPTED_LICENSE_BODY decrypted with LICENSE_KEY & IV (AES/CBC/PKCS5Padding)

Can this be done with crypto-js?

Thanks very much for any help

**Comment by user163** [comment]
No. `RSA/ECB/PKCS1Padding` is asymmetric cryptography. CryptoJS is a library for symmetric cryptography and does not support asymmetric cryptography.   

----------

Incidentally, `RSA/ECB/PKCS1Padding` only allows **_either_** encryption with the public key / decryption with the private key **_or_** signing with the private key / verification with the public key.  
Encrypting with the private key / decrypting with the public key is not defined at all. Some libraries implement this as low level signing/verifying  (e.g. Java's SunJCE provider, which also uses the identifier `RSA/ECB/PKCS1Padding`, so that the low level signature may have been created with it).  
However, this has nothing to do with encryption/decryption in the actual sense, since decryption with the public key can never create confidentiality (by definition, anyone can have the public key, i.e. anyone can decrypt).
